jobbergateapi2/tests/apps/users/test_routers.py

```diff
@@ -204,3 +204,95 @@ async def test_users_me(client, user_data):
 
     data = response.json()
     assert data["email"] == user_data["email"]
+
+
+@pytest.mark.asyncio
+@database.transaction(force_rollback=True)
+async def test_update_user(client, user_data):
+    """
+    Test update an User.
+    """
+    user = [UserCreate(id=1, **user_data)]
+    await insert_objects(user, users_table)
+
+    response = client.put("/users/1", json={"full_name": "new name"})
+
+    assert response.status_code == status.HTTP_201_CREATED
+    data = response.json()
+
+    assert data["full_name"] == "new name"
+    assert data["id"] == 1
+
+    query = users_table.select(users_table.c.id == 1)
+    user = User.parse_obj(await database.fetch_one(query))
+
+    assert user is not None
+    assert user.full_name == "new name"
+
+
+@pytest.mark.asyncio
+@database.transaction(force_rollback=True)
+async def test_update_user_superuser(client, user_data):
+    """
+    Test update an User to change the superuser status.
+    """
+    user = [UserCreate(id=1, is_superuser=True, **user_data)]
+    await insert_objects(user, users_table)
+
+    response = client.put("/users/1", json={"is_superuser": False})
+
+    assert response.status_code == status.HTTP_201_CREATED
+    data = response.json()
+
+    assert data["is_superuser"] is False
+    assert data["id"] == 1
+
+    query = users_table.select(users_table.c.id == 1)
+    user = User.parse_obj(await database.fetch_one(query))
+
+    assert user is not None
+    assert user.is_superuser is False
+
+
+@pytest.mark.asyncio
+@database.transaction(force_rollback=True)
+async def test_update_user_not_found(client, user_data):
+    """
+    Try to update a User not found, must return 404 and do nothing with the stored data.
+    """
+    user = [UserCreate(id=1, **user_data)]
+    await insert_objects(user, users_table)
+
+    response = client.put("/users/123", json={"full_name": "new name"})
+
+    assert response.status_code == status.HTTP_404_NOT_FOUND
+
+    query = users_table.select(users_table.c.id == 1)
+    user = User.parse_obj(await database.fetch_one(query))
+
+    assert user is not None
+    assert user.full_name == user_data["full_name"]
+
+
+@pytest.mark.asyncio
+@database.transaction(force_rollback=True)
+async def test_update_user_me(client, user_data):
+    """
+    Test update the authenticated User.
+    """
+    user = [UserCreate(id=1, **user_data)]
+    await insert_objects(user, users_table)
+
+    response = client.put("/user/me", json={"full_name": "new name"})
+    data = response.json()
+
+    assert data["full_name"] == "new name"
+    assert data["id"] == 1
+
+    assert response.status_code == status.HTTP_201_CREATED
```

Comment:
should we use HTTP 204 to imply "resource updated successfully"?

Reply:
No, because we are indeed returning a body